### DISCUSSION TITLE
Feature planning for enabling conditional generative methods in MRI reconstruction

### DISCUSSION THREAD

**Discussion by cxlcl** [Project-MONAI/MONAI#7161] [Ideas]
Hi everyone, 

The goal for the new features is to include the building blocks of Diffusion Model based MRI reconstruction and test-time hyper-parameter tuning, as implemented in [this research repo](https://github.com/NVlabs/SMRD). The new features would be added to MONAI-core. Based on the discussion with @wyli,  those new features would be: 

* the SURE loss function (line 6 in the SMRD algorithm in the link above)
* the generation step using pre-trained score function (line 3 in the algorithm)
* the conjugate gradient step to enforce measurement consistency (line 4 in the algorithm)

 The plan is to first implement the SURE loss and then the conjugate gradient step. For the generation step, we may already have good implementations in this[ Generative Models repo](https://github.com/Project-MONAI/GenerativeModels/tree/main) in MONAI. I will look into the inference module in the this repo, and start from there if new implementation is needed.

Please feel free to comment.
